### ISSUE TITLE
One time urls? Disable reconnecting?

### ISSUE THREAD

**Issue by ursusursus** [Tinder/Scarlet#69]
Hi, my backend generates a wss:// url for me, **which however is a one time thing**, meaning upon connetion lost, I wont be able to reconnect, which is what Im seeing with Scarlet, (infinite loop of WaitingToRetry, Connecting)

Is there way to disable the reconnecting policy and Ill maybe handle connecting back manually? Since I need to call backend api again to generate a new wss:// url

Thanks

**Comment by hantrungkien** [comment]
Yes. You can try my lib, it's config with Scartlet while waiting for my Pull request to be approved.
https://github.com/hantrungkien/tinder-scarlet-lifecycle-android
Its purpose is to make sure Scartlet completes when there is a LifecycleRegistry event completed. You must have custom one LifecycleRegistry like a LoggedInLifecycle in demo project.

> @aenonGit @raylee4204 Yes it is possible. The WebSocket [echo demo](https://github.com/Tinder/Scarlet/tree/0.2.x/demo/src/main/java/com/tinder/app/websocket/echo) has a Login/Logout button that turns the connection on and off.
> 
> The trick is to define a [custom lifecycle](https://github.com/Tinder/Scarlet/blob/0.2.x/demo/src/main/java/com/tinder/app/websocket/echo/domain/LoggedInLifecycle.kt) using the [LifecycleRegistry](https://github.com/Tinder/Scarlet/blob/0.2.x/scarlet/src/main/java/com/tinder/scarlet/lifecycle/LifecycleRegistry.kt) class and have the UI [modify it](https://github.com/Tinder/Scarlet/blob/0.2.x/demo/src/main/java/com/tinder/app/websocket/echo/view/EchoBotViewModel.kt#L39)
